stl/inc/mdspan

```diff
@@ -0,0 +1,901 @@
+
+    constexpr default_accessor() noexcept = default;
+
+    template <class _OtherElementType,
+        enable_if_t<
+            is_convertible_v<typename default_accessor<_OtherElementType>::element_type (*)[], _ElementType (*)[]>,
+            int> = 0>
+    constexpr default_accessor(default_accessor<_OtherElementType>) noexcept {}
+
+    _NODISCARD constexpr data_handle_type offset(data_handle_type _Ptr, size_t _Idx) const noexcept {
+        return _Ptr + _Idx;
+    }
+
+    _NODISCARD constexpr reference access(data_handle_type _Ptr, size_t _Idx) const noexcept {
+        return _Ptr[_Idx];
+    }
+};
+
+template <class _ElementType, class _Extents, class _LayoutPolicy = layout_right,
+    class _AccessorPolicy = default_accessor<_ElementType>>
+class mdspan {
+public:
+    // Domain and codomain types
+    using extents_type     = _Extents;
+    using layout_type      = _LayoutPolicy;
+    using accessor_type    = _AccessorPolicy;
+    using mapping_type     = typename layout_type::template mapping<extents_type>;
+    using element_type     = _ElementType;
+    using value_type       = remove_cv_t<element_type>;
+    using index_type       = typename extents_type::index_type;
+    using size_type        = typename extents_type::size_type;
+    using rank_type        = typename extents_type::rank_type;
+    using data_handle_type = typename accessor_type::data_handle_type;
+    using reference        = typename accessor_type::reference;
+
+    _NODISCARD static constexpr rank_type rank() noexcept {
+        return _Extents::rank();
+    }
+    _NODISCARD static constexpr rank_type rank_dynamic() noexcept {
+        return _Extents::rank_dynamic();
+    }
+    _NODISCARD static constexpr size_t static_extent(const rank_type r) noexcept {
+        return _Extents::static_extent(r);
+    }
+
+    template <class _Mapping = mapping_type,
+        enable_if_t<(rank_dynamic() > 0) && is_default_constructible_v<data_handle_type>
+                        && is_default_constructible_v<_Mapping> && is_default_constructible_v<accessor_type>,
+            int>             = 0>
+    constexpr mdspan() {}
+    constexpr mdspan(const mdspan& rhs) = default;
+    constexpr mdspan(mdspan&& rhs)      = default;
+
+    template <class _Mapping = mapping_type,
+        enable_if_t<(rank() == 0 || rank_dynamic() == 0)
+                        && is_constructible_v<_Mapping, extents_type> && is_default_constructible_v<accessor_type>,
+            int>             = 0>
+    explicit constexpr mdspan(data_handle_type _Ptr_) : _Ptr{_STD move(_Ptr_)}, _Map{extents_type{}} {}
+
+    template <class... _OtherIndexTypes,
+        enable_if_t<((sizeof...(_OtherIndexTypes) > 0)) && (is_convertible_v<_OtherIndexTypes, index_type> && ...)
+                        && (is_nothrow_constructible_v<index_type, _OtherIndexTypes> && ...)
+                        //&& (sizeof...(_OtherIndexTypes) == rank() || sizeof...(_OtherIndexTypes) == rank_dynamic())

```

Comment:
This is commented out because it breaks one of the tests,
```cpp
static constexpr int arr[6] = {};
constexpr mdspan<const int, extents<size_t, dynamic_extent, 3>> mds1(arr, 2);
```
but I haven't been able to figure out why.